text/0000-default-fields.md

```diff
@@ -0,0 +1,305 @@
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Field defaults are limited to constant expressions. This means there are values that can't be used as defaults, so any value that requires allocation, including common collections like `Vec::new()`. It's expected that users will use a constructor function or builder for initialisers that require allocations.
+
+# Alternatives
+[alternatives]: #alternatives
+
+## Allow arbitrary expressions instead of just constant expressions
+
+Allowing arbitrary expressions as field defaults would make this feature more powerful. However, limiting field defaults to constant expressions maintains the expectation that struct literals are cheap and deterministic. The same isn't true when arbitrary expressions that could reasonably panic or block on io are allowed.
+
+For complex initialisation logic, builders are the preferred option because they don't carry this same expectation.
+
+## Allow `Default::default()` instead of just constant expressions
+
+An alternative to allowing any expression as a default value is allowing `Default::default()`, which is expected to be cheap and deterministic:
+
+```rust
+struct Foo {
+    a: &'static str,
+    b: Vec<i32> = Vec::default(),
+    c: i32,
+}
+
+let foo = Foo {
+    a: "Hello",
+    c: 42,
+}
+```
+
+It could be argued that supporting `default()` is an artificial constraint that doesn't prevent arbitrary expressions. The difference is that `Default` has an expectation of being cheap, so using it to inject logic into field initialisation is an obvious code smell.
+
+Allowing functionality to be injected into data initialisation through `default()` means struct literals may have runtime costs that aren't ever surfaced to the caller. This goes against the expectation that literal expressions have small, predictable runtime cost and are totally deterministic (and trivially predictable).
+
+## Type inference for fields with defaults
+
+Reduce the amount of code needed to define a `struct` by allowing type inference for field defaults:
+
+```rust
+struct Foo {
+    a: &'static str,
+    b = Bar,
+    c: i32,
+}
+```
+
+This has the effect of simplifying the definition, but also requiring readers to manually work out what the type of the right-hand-side is. This could be less of an issue with IDE support.
+
+## Explicit syntax for opting into field defaults
+
+Field defaults could require callers to use an opt-in syntax like `..`. This would make it clearer to callers that additional code could be run on struct initialisation, weakening arguments against more powerful default expressions. However it would prevent field default from being used to maintain backwards compatibility, and reduce overall ergonomics. If it's unclear whether or not a particular caller will use a default field value then its addition can't be treated as a non-breaking change. 
```

Comment:
>However it would prevent field default from being used to maintain backwards compatibility

There's no backward compatibility already (https://github.com/rust-lang/rfcs/pull/1806/files#r90957633), so mirroring pattern syntax or omitting `..` is mostly a stylistic choice (unless, again, `..` is made optional in patterns).